packages/components/src/avatar/avatar.tsx

```diff
@@ -1,97 +1,356 @@
+}
+
+const radiusSizes: Record<
+  NonNullable<AvatarProps['size']>,
+  number | undefined
+> = {
+  '80': 16,
+  '56': undefined,
+  '48': 12,
+  '32': 10,
+  '28': 8,
+  '24': 8,
+  '20': 6,
+  '16': undefined,
+}
+
+const emojiSizes: Record<AvatarProps['size'], TextProps['size'] | undefined> = {
+  '80': 32,
+  '56': undefined,
+  '48': undefined,
+  '32': 15,
+  '28': undefined,
+  '24': 13,
+  '20': 11,
+  '16': undefined,
+}
+
+const textSizes: Record<NonNullable<AvatarProps['size']>, TextProps['size']> = {
+  '80': 32,
+  '56': 27,
+  '48': 19,
+  '32': 15,
+  '28': 13,
+  '24': 13,
+  '20': 11,
+  '16': 11,
+}
+
+const iconSizes: Record<
+  NonNullable<AvatarProps['size']>,
+  React.ReactElement
+> = {
+  '80': <MembersIcon20 />,
+  '56': <></>,
+  '48': <MembersIcon20 />,
+  '32': <MembersIcon16 />,
+  '28': <MembersIcon16 />,
+  '24': <></>,
+  '20': <MembersIcon12 />,
+  '16': <></>,
+}
+
+function hasIdenticon(
+  props: AvatarProps
+): props is Extract<AvatarProps, { colorHash?: number[][] }> {
+  return (
+    (props as Extract<AvatarProps, { colorHash?: number[][] }>).colorHash !==
+    undefined
+  )
+}
+
+function hasLock(
+  props: AvatarProps
+): props is Extract<AvatarProps, { lock?: 'locked' | 'unlocked' }> {
+  return (
+    (props as Extract<AvatarProps, { lock?: 'locked' | 'unlocked' }>).lock !==
+    undefined
+  )
+}
+
+function hasImage(
+  props: AvatarProps
+): props is Extract<AvatarProps, { src?: string }> {
+  return (props as Extract<AvatarProps, { src?: string }>).src !== undefined
+}
+
+const Avatar = (props: AvatarProps) => {
+  const { type, size, backgroundColor = '$neutral-95', outline = false } = props
+
+  // todo?: conditional hook(s)
+
+  const colorHash = hasIdenticon(props) ? props.colorHash : null
+  const identiconRing = useMemo(() => {
+    if (colorHash) {
+      const gradient = generateIdenticonRing(colorHash)
+      return `conic-gradient(from 90deg, ${gradient})`
+    }
+  }, [colorHash])
 
   const [status, setStatus] = useState<ImageLoadingStatus>('idle')
 
-  useEffect(() => {
-    setStatus('idle')
-  }, [src])
+  const padding = identiconRing ? paddingSizes[size] : 0
+  const radius =
+    type === 'account' ? radiusSizes[size] : tokens.radius['full'].val
+  const src = hasImage(props) ? props.src : null
 
   return (
-    <Base size={size} shape={shape} outline={outline}>
-      {indicator !== 'none' && (
+    <Stack style={{ position: 'relative' }}>
+      <Base
+        borderRadius={radius}
+        padding={padding}
+        size={size}
+        outline={outline}
+        backgroundColor={backgroundColor}
+        // todo?: https://reactnative.dev/docs/images.html#background-image-via-nesting or svg instead
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        style={{
+          ...(Platform.OS === 'web' && {
+            background: identiconRing,
+          }),
+        }}
+      >
+        {/* todo?: render fn(s) */}
```

Comment:
I would do that.

Reply:
Done at 3797b4cfb857067215c4b732dddf8f71fadc15ce with `renderContent` and `renderBadge`. @jkbktl would you do it like that?

Reply:
It's very well done. Ternary ('group') there seems very hard to read, but I was not able to figure out better solution, so let's keep it like this for now. Good job.